services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/allu/CableReportServiceAllu.kt

```diff
@@ -281,6 +273,33 @@ class CableReportServiceAllu(
             .orElseThrow()
     }
 
+    private fun postAttachment(alluId: Int, token: String, attachment: Attachment) {
+        logger.info { "Sending attachment for application $alluId." }
+
+        val builder = MultipartBodyBuilder()
+        builder
+            .part("metadata", attachment.metadata, MediaType.APPLICATION_JSON)
+            .filename("metadata")
+        builder
+            .part("file", ByteArrayResource(attachment.file), MediaType.APPLICATION_PDF)
```

Comment:
We'll be sending other attachments besides PDFs now. The media type should be read from `attachment.metadata.mimeType`.

Reply:
Changed as suggested. Currently the junit test does not verify the content type. Actually there wasn't any junit tests for the whole method I think. This can be added but testing multipart content seems to be overly complicated.

Reply:
Seems like it could be checked with:
```
        val boundary = request.headers["Content-Type"]?.split(";boundary=")!![1]
        val reader = MultipartReader(request.body, boundary)

        val types = buildList {
            while (true) {
                val part = reader.nextPart() ?: break
                add(part.headers["Content-Type"])
            }
        }
        
        assertThat(types).containsExactly("application/json", "image/png")
```
Where everything but the assert could be in a private function.

Reply:
Added content type check.